RedHat-UX/red-hat-design-system
# feat(button): add <rh-button>

## What I did

1. Add rh-button ~~based on BaseButton from pfe~~
2. Uses the new ElementInternals API to remove need for slotted button

Closes #370 

## Testing Instructions

1. `npm start`

And check the DP demos:

4. [Main demo](https://deploy-preview-550--red-hat-design-system.netlify.app/components/button/demo)
5. [Color context demo](https://deploy-preview-550--red-hat-design-system.netlify.app/components/button/demo/color-context/)
6. [Form control demo](https://deploy-preview-550--red-hat-design-system.netlify.app/components/button/demo/form-control/)

## Notes to Reviewers
